include/eve/detail/function/simd/riscv/subscript.hpp

```diff
@@ -0,0 +1,112 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/rvv_common_masks.hpp>
+
+namespace eve::detail
+{
+template<typename T, typename N>
+EVE_FORCEINLINE T
+extract(wide<T, N> const& v, std::size_t i) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  auto OnFirstNeeded = __riscv_vslidedown_tu(v, v, i, N::value);
+  if constexpr( std::is_floating_point_v<T> ) return __riscv_vfmv_f(OnFirstNeeded);
+  else return __riscv_vmv_x(OnFirstNeeded);
+}
+
+template<typename T, typename N>
+EVE_FORCEINLINE void
+insert(wide<T, N>& v, std::size_t i, std::convertible_to<T> auto x) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  // get mask with 1 on i'th element.
+  auto mask = rvv_one<T, N>(i);
+  v         = if_else(mask, static_cast<T>(x), v);
+}
+
+template<typename T, typename N>
+EVE_FORCEINLINE logical<T>
+                extract(logical<wide<T, N>> const                &v, std::size_t i) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  using u_m1_cardinal               = typename riscv_rvv_dyn_::m1_fixed<unsigned>;
+  using inter_type                  = wide<unsigned, u_m1_cardinal>;
+  auto             uint_wide        = bit_cast(v, as<inter_type> {});
+  constexpr size_t sew              = sizeof(unsigned) * 8;
+  size_t           expected_element = i / sew;
+  auto             val              = extract(uint_wide, expected_element);
+  bool             to_ret           = std::bitset<sew>(val)[i % sew];
```

Comment:
don't use std::bitset, it's pretty bad.

Reply:
Removed as now it uses store/load from stack.